core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -65,6 +66,26 @@ public Table loadTable(TableIdentifier identifier) {
     return result;
   }
 
+  @Override
+  public Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
+    Preconditions.checkArgument(
+        (identifier != null) && isValidIdentifier(identifier), "Invalid identifier: %s", identifier);
+    Preconditions.checkArgument(metadataFileLocation != null && !metadataFileLocation.isEmpty(),
+        "Cannot register an empty metadata file location as a table");
+
+    // Throw an exception if this table already exists in the catalog.
```

Comment:
I think it makes sense to adjust the impl here to what Ryan suggested in https://github.com/apache/iceberg/pull/4099/files#r805404106

Reply:
A similar comment was posted out by @ajantha-bhat, please refer https://github.com/apache/iceberg/pull/5037#discussion_r898211187
TLDR it gives an error